### ISSUE TITLE
Create SASS version

### ISSUE THREAD

**Issue by kristoferjoseph** [kristoferjoseph/flexboxgrid#41]
Because for some reason people love complicating their build script. ;)

**Comment by chorijan** [comment]
+1 for the SASS version. any idea for a ETA?

**Comment by dnomak** [comment]
try it please https://github.com/flexiblegs/flexiblegs/tree/master/sass CodePen http://codepen.io/dnomak/pen/wBzdzj

**Comment by einthusan** [comment]
I see that the less version if there also, can be it used in production?

**Comment by kristoferjoseph** [comment]
@chorijan Soon? Was going to push something this weekend.
@einthusan I hope I am understanding what you are asking. The plan is to make production ready versions of those preprocessors listed.

**Comment by einthusan** [comment]
@kristoferjoseph okay thanks. I just wasn't sure if the production version was ready or not, but by seeing that the issue is still open, i realized it's still a work in progress.

Thanks for all your hard work!

**Comment by suenot** [comment]
+1, I needed in rebuild grid to 24 collumns.

**Comment by kristoferjoseph** [comment]
Worked on it a bit today. Will post it in a branch tomorrow.

**Comment by benmliang** [comment]
+1 awesome job

**Comment by kristoferjoseph** [comment]
Here are the beginnings of the SASS version: https://github.com/kristoferjoseph/flexboxgrid/pull/55/files

It is not working yet, but I wanted to share the direction I was heading with all interested parties. 
The other flavors will be based off of this eventually.

**Comment by guillaumevincent** [comment]
awesome feature, how can people help ?

**Comment by kristoferjoseph** [comment]
@guillaumevincent check out that branch and poke around. Feedback welcome. :)

**Comment by kristoferjoseph** [comment]
@GLStephen @stereokai @chorijan @einthusan @suenot @benmliang @guillaumevincent and @SlexAxton 

PR #55 is ready for feedback. It is "Feature Complete" so it could be pushed to master, but I'd love a sanity check on where it is at currently. The output is correct and the mixins I could see being useful on their own. 
Only part I dislike is the [auto-width](https://github.com/kristoferjoseph/flexboxgrid/pull/55/files#diff-7fdd5c4c82208e16efd2f16ee7d18c95R40) column function. Seems a bit arbitrary. If you can think of a better way to do that have at it.
Giving every class a responsive modifier has grown the file size larger than I'd like, but I feel like we could add a way to only add the properties you plan on using. 
One step at a time. :)

**Comment by SlexAxton** [comment]
I don't know a ton about SASS or what people need explicitly, so ignore my feedback but...

this seem p good to me.

**Comment by GLStephen** [comment]
At a glance it looks good. I will integrate into our SCSS flow this week and see what it emits.

**Comment by kristoferjoseph** [comment]
Thank @GLStephen & @SlexAxton

**Comment by ffub** [comment]
#55 works well for me. I have a few thoughts on including flexboxgrid as part of a larger project.

As a module I'd rather an import didn't output any code until I run a mixin. You could have a master mixin called flexboxgrid that does that, and into which you could pass basic config. Likewise it would be good to use '!default' on any global variables.

**Comment by kristoferjoseph** [comment]
@ffub thanks for taking a look. Really appreciated. I will add your feedback to #55.

**Comment by gonsfx** [comment]
Does not compile for me. What sass version should i be using and would you mind adding it to a Gemfile?

**Comment by kristoferjoseph** [comment]
@gonsfx You can look at the [package.json](https://github.com/kristoferjoseph/flexboxgrid/blob/sass/package.json) to see what version I am using for the build. It uses [gulp-sass](https://github.com/dlmanning/gulp-sass/blob/master/package.json) which uses node-sass...

This is one of the reasons I didn't add a SASS version to begin with. Maintenance can be difficult. I can really only support the way I build this project ( which uses gulp ) as I would only be guessing as to the myriad other ways people are building in the wild.

I don't use `Gemfiles` so I don't have the insight needed, but if you can get it to work, please make a pull request as I am sure lot's of others will run into the issue you are having.

Thank you very much :)

**Comment by gonsfx** [comment]
Don't get me wrong - i absolutely love the sass port and think it should be included in `master` better sooner than later.

Nevertheless, compiling did not work. If you choose to add it to `master`, be sure to explain that `node-sass` is used instead of the ruby gem and that those have different features.

On a sidenote, you have an `npm-shrinkwrap.json` next to your `package.json` so installing from your projects root directory with `npm install` installs the exact same versions that you used.

Problem is that the `gulp-sass` version you shrinkwrapped is `1.3.2` which in turn uses `node-sass` version `2.0.0-beta`. When running one of the current node stable versions `0.12.x`, your project doesn't compile. To compile with the newer node stable versions, `node-sass` is needed in version `2.0.1`.

**Comment by kristoferjoseph** [comment]
@gonsfx Wow, thanks for pointing that issue out. I added that shrink wrap file to try and fix the possible issue, seems it made an issue of it's own. I run update and see if it fixes it. 

When you say compiling did not work, are you saying when you run `gulp` it doesn't build for you?

**Comment by gonsfx** [comment]
Yes, that is correct. Running `gulp` threw errors.

**Comment by kristoferjoseph** [comment]
@gonsfx Please pull and let me know if you still see errors. Thanks.

**Comment by gonsfx** [comment]
Seems to work. Switched back from our fork to your #sass branch

**Comment by kristoferjoseph** [comment]
@gonsfx Awesome! Was wondering if you have any feedback on the proposal made [in the LESS issue](https://github.com/kristoferjoseph/flexboxgrid/issues/42#issuecomment-102157915) about project structure to assist in importing mixins etc. 

Ideally I would have something similar for each preprocessor. 

Thanks again for the feedback.

**Comment by cmoralesweb** [comment]
Just a small note, in case you didn't know it:

```gulp-sass``` uses ```node-sass```, but there is a ruby plugin available (```gulp-ruby-sass```) which uses the official library.

**Comment by kristoferjoseph** [comment]
@cmoralesweb Thanks. Please send a PR once 7.0 is released. I have moved away from gulp for the next release though due to all of the library version mismatches.

**Comment by cmoralesweb** [comment]
Moved to grunt or to nothing at all? I can help with grunt/gulp if needed. 

I am tweaking the library so it generates %placeholders for the grid, in case you are interested. Similar to what I did here: https://github.com/cmoralesweb/atomic-foundation/blob/master/foundation-placeholders/_grid-placeholders.scss

**Comment by kristoferjoseph** [comment]
@cmoralesweb I have switched to npm run scripts. Much less headache involved. Grunt/Gulp scripts are over kill for simply running a file through a preprocessor. The overhead only begins to make sense at the scale of an application. input > output is a very simple thing to do in a run script and it has no dependencies to struggle with keeping up to date.

Placeholders are cool, not sure that is a use case I want to support. Too much room for error on the user side leading to a possible ton of issues filed. It would be akin to a foot cannon mixing in grid properties into other selectors invisibly.

**Comment by gonsfx** [comment]
@cmoralesweb @kristoferjoseph
node-sass is a lot faster than ruby-sass so i would encourge to stay with it, if it compiles (which it does).
i don't see a reason to switch to ruby-sass.

**Comment by cmoralesweb** [comment]
@gonsfx @kristoferjoseph 

Basically this: https://github.com/sass/libsass/wiki/The-LibSass-Compatibility-Plan

node-sass is faster than ruby-sass, indeed, but it lacks some features from the official library, mainly full placeholder support. 

Regarding the "it compiles"... it does in node-sass, not in ruby-sass. I tried using it with ruby-sass (3.4) and it failed. I need ruby-sass in my development (I use placeholders a lot), so I can't import the grid along with the rest of my files, and it's not viable to have node-sass on one side and ruby-sass on the other.

**Comment by ffub** [comment]
Just replace the use of extend with include for the columns. You’ll get a few extra lines of CSS, but SASS doesn’t allow extending within media queries.

**Comment by kristoferjoseph** [comment]
@cmoralesweb thank you for the info, I appreciate it. Just so we are all on the same page, I will not be switching to ruby-sass nor will i be making placeholders for this grid. With that said, if you want to file a separate issue with the specific error you get when you compile with ruby-sass I can see if there is a change I can make to my code to at least avoid the error. Thanks!

**Comment by cmoralesweb** [comment]
@kristoferjoseph I think @ffub nailed the error, so no need for a new issue, I guess :smile: 

I understand your decision about ruby/placeholders :+1:

**Comment by kristoferjoseph** [comment]
@cmoralesweb thank you for understanding. Been really trying to limit scope so I can ship this dang release that has been taking fo'evah. We can see if people are needing placeholders etc. after 7.0 drops and we can reconsider. Thanks again for being chill bud.

**Comment by cmoralesweb** [comment]
Just mention me if you ever implement them and need my help.

**Comment by kristoferjoseph** [comment]
@cmoralesweb :+1: will do :heart:

**Comment by miraage** [comment]
@kristoferjoseph , please, keep in mind some things.
It would be great too see mixins, so we can use your components in our projects. For example:

```sass
// Somewhere in flexboxgrid-filename.scss
@mixin make-column($count) {
  // column definition
}

// Somewhere in our projects
.article__controls {
  @include make-column(6);
}
```

**Comment by kukac7** [comment]
:+1:

**Comment by Madebym** [comment]
+1 @miraage

**Comment by fabianmarz** [comment]
:+1:

**Comment by dnomak** [comment]
@kristoferjoseph please check out https://github.com/flexiblegs/flexiblegs-scss

**Comment by Romainpetit** [comment]
What's missing now ? The Sass branch works like a charm here. It's just weird to see Grunt being switched to Gulp

**Comment by kristoferjoseph** [comment]
@Romainpetit There are still some incompatibilities with sass versions and compilers. The switch to gulp was done at the time to use the most compatible plugin. Just a note, not interested in debate over this change. Thanks.

**Comment by cmoralesweb** [comment]
@kristoferjoseph Hey, just a note, in case you are interested. Libsass has released 3.3 (http://robotsaldetalle.es/pruebas/) and it (finally!!) supports '%placeholders' properly. So my concerns in my previous comment (https://github.com/kristoferjoseph/flexboxgrid/issues/41#issuecomment-110724838) are no longer valid. 

I know you never intended to switch to ruby, but still I thought it would be nice to point it out :)

**Comment by kristoferjoseph** [comment]
I will not be supporting any preprocessors. The workflow will be CSS only. 7.0 was stalled for way too long over preprocessor issues and I want this project to continue to look forward. Apologies.
